### PR TITLE
MEN-4332: Redact sensitive information from logs

### PR DESCRIPTION
**Pull request by alfrunes**
Prevents a potential email leakage from the logs.

- [x] Blocked by mendersoftware/useradm-enterprise#188

### CODE REVIEW

File: client/tenant/client_tenantadm.go [hunk 1]
```diff
@@ -133,10 +133,13 @@ func (c *Client) CheckHealth(ctx context.Context) error {
 }
 
 func (c *Client) GetTenant(ctx context.Context, username string, client apiclient.HttpRunner) (*Tenant, error) {
+	usernameQ := url.QueryEscape(username)
 	req, err := http.NewRequest(http.MethodGet,
 		JoinURL(c.conf.TenantAdmAddr, GetTenantsUri+"?username="+url.QueryEscape(username)),
 		nil)
 	if err != nil {
+		repl := strings.NewReplacer(username, "REDACTED", usernameQ, "REDACTED")
```
Comment (merlin-northern) [inline client/tenant/client_tenantadm.go:141]: can we have `const Redacted = "REDACTED"` or similar?

File: client/tenant/client_tenantadm.go [hunk 2]
```diff
@@ -133,10 +133,13 @@ func (c *Client) CheckHealth(ctx context.Context) error {
...
+		err = errors.New(repl.Replace(err.Error()))
```
Comment (merlin-northern) [inline client/tenant/client_tenantadm.go:142]: consider changing into `ReplaceAll`?

File: client/tenant/client_tenantadm.go [hunk 3]
```diff
@@ -145,6 +148,8 @@ func (c *Client) GetTenant(ctx context.Context, username string, client apiclien
 
 	rsp, err := client.Do(req.WithContext(ctx))
 	if err != nil {
+		repl := strings.NewReplacer(username, "REDACTED", usernameQ, "REDACTED")
+		err = errors.New(repl.Replace(err.Error()))
```
Comment (merlin-northern) [inline client/tenant/client_tenantadm.go:152]: same here, consider `ReplaceAll`?

Comment (alfrunes) [reply client/tenant/client_tenantadm.go:152]: `ReplaceAll` does not support multiple replacement pairs in one call.

**Review by tranchitella — APPROVED** [review]

**Pull request merged**